### ISSUE TITLE
Pageable.unpaged() created inside AggregatedPageImpl causes UnsupportedOperationException when mapping to PagedResources

### ISSUE THREAD

**Issue by aminmc** [VanRoy/spring-data-jest#49]
Hi @VanRoy 

I've graded to the latest version on the library and have noticed the following problem:
I have some code that does a search and returns a list of objects and then I have spring Hateoas to convert the Paged objects into PagedResources.  However with the latest update I get the following exception:
```

java.lang.UnsupportedOperationException: null
	at org.springframework.data.domain.Unpaged.getPageNumber(Unpaged.java:87)
	at org.springframework.data.web.HateoasPageableHandlerMethodArgumentResolver.enhance(HateoasPageableHandlerMethodArgumentResolver.java:116)
```

This is because:
```
public <T> AggregatedPage<T> mapResults(SearchResult response, Class<T> clazz, List<AbstractAggregationBuilder> aggregations) {

		LinkedList<T> results = new LinkedList<>();

		for (SearchResult.Hit<JsonObject, Void> hit : response.getHits(JsonObject.class)) {
			if (hit != null) {
				results.add(mapSource(hit.source, clazz));
			}
		}

		String scrollId = null;
		if (response instanceof ExtendedSearchResult) {
			scrollId = ((ExtendedSearchResult) response).getScrollId();
		}

		return new AggregatedPageImpl<>(results, response.getTotal(), response.getAggregations(), scrollId);
	}
```
The aggregatedPageImpl creates a Pageable.unpaged instance. The api I'm using to do the search is 
```
repository.search(searchQuery)
```

In the DefaultResultMapper in Spring they do:
```
	@Override
	public <T> AggregatedPage<T> mapResults(SearchResponse response, Class<T> clazz, Pageable pageable) {
		long totalHits = response.getHits().totalHits();
		List<T> results = new ArrayList<>();
		for (SearchHit hit : response.getHits()) {
			if (hit != null) {
				T result = null;
				if (StringUtils.isNotBlank(hit.sourceAsString())) {
					result = mapEntity(hit.sourceAsString(), clazz);
				} else {
					result = mapEntity(hit.getFields().values(), clazz);
				}
				setPersistentEntityId(result, hit.getId(), clazz);
				populateScriptFields(result, hit);
				results.add(result);
			}
		}

        return new AggregatedPageImpl<T>(results, **pageable**, totalHits, response.getAggregations(), response.getScrollId());
	}
```
Is there a work around for this? This used to work in version 2.3.x (my AWS ES version is 2.x)

Thanks

**Comment by VanRoy** [comment]
Hi @aminmc , thanks for the report. It's appear that @shurth fixed this issue in this PR : #48 . I will quickly check and integrate it in the next release.

**Comment by aminmc** [comment]
Thanks! Sorry I wasn’t aware a PR had already been raised. Look forward to the next release. Do you have a timeframe for the release?

**Comment by mancini0** [comment]
Thank you aminmc and VanRoy, your efforts are much appreciated.

**Comment by VanRoy** [comment]
The release is planned just after the SpringBoot 2.0

**Comment by VanRoy** [comment]
\o/ Spring Boot 2 released, I just released Spring Data Jest 3.1.0.
